### DISCUSSION TITLE
How to set up options to create user in MongoDB

### DISCUSSION THREAD

**Discussion by SirasornT** [nextauthjs/next-auth#899] [Help]
**Your question**
Is next-auth supposed to create an entry in MongoDB automatically after a user logs in? If so, how do I set it up?

**What are you trying to do**
I am trying to create a user in the `users` collection in MongoDB after the user logs in with Google for the first time. I have tried connecting my database to the example project but my MongoDB atlas collection was still empty after the user logs in.
I have been stuck on this for days and couldn't figure out how to do it properly. Right now, I am manually creating the user in the `signIn` event but the problem is that the user being created will have a different _id from the user.id in the `session` callback. 

```javascript
const options = {
  site: process.env.SITE,
  providers: [
    Providers.Google({
      clientId: process.env.GOOGLE_ID,
      clientSecret: process.env.GOOGLE_SECRET,
      authorizationUrl:
        "https://accounts.google.com/o/oauth2/v2/auth?prompt=consent&access_type=offline&response_type=code",
      accessTokenUrl: "https://oauth2.googleapis.com/token",
    }),
  ],
  database: {
    type: "mongodb",
    uri: process.env.DATABASE_URL,
    w: "majority",
    useNewUrlParser: true,
    useUnifiedTopology: true,
    retryWrites: true,
  },
  debug: false,
  events: {
    signIn: async (message) => {
      const { name, email, image } = message.user;
      const { client } = await connectToDatabase();

      let usersCollection = client
        .db(process.env.DATABASE_NAME)
        .collection("users");

      try {
        let user = await usersCollection.findOne({ email });

        if (user) {
          console.log("user already exist");
        } else {
          await usersCollection.insertOne({ name, email, image });
        }
      } catch (err) {
        console.log("Server error");
        console.log(err);
      }
    },
  },
  callbacks: {
    session: async (session, user) => {
      session.id = user.id;
      return Promise.resolve(session);
    },
  },
};
```

**Feedback**
I think I should be setting up adapters for MongoDB but I am not sure. I am new to working with databases and authentication so I think having an example specific to MongoDB (just like the Prisma example) would be nice. Thank you so much!

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by eikomaniac** [comment]
Hi, did you manage to figure this out? I also don't know what the `persist database` option does compared to the JWT option...
